File: src/main/resources/templates/java/solution/src/${packageNameFolder}/Context.java
package ${packageName};

import java.util.Date;
import java.util.List;

public class Context {

	private SortStrategy sortAlgorithm;

	private List<Date> dates;

	public List<Date> getDates() {
		return dates;
	}

	public void setDates(List<Date> dates) {
		this.dates = dates;
	}

	public void setSortAlgorithm(SortStrategy sa) {
		sortAlgorithm = sa;
	}

	public SortStrategy getSortAlgorithm() {
		return sortAlgorithm;
	}

	public void sort() {
		sortAlgorithm.performSort(this.dates);
	}
}

File: src/main/resources/templates/kotlin/solution/src/${packageNameFolder}/Context.kt
package ${packageName}

class Context {

    //TODO: add the missing attributes
    //TODO: add the missing methods
    /*<remove>*/
    var sortAlgorithm: SortStrategy? = null
    var array: Array<Int>? = null

    fun sort() {
        val arr = this.array ?: error("No array specified!")
        val sortAlgorithm = this.sortAlgorithm ?: error("No sort algorithm specified!")
        sortAlgorithm.performSort(arr)
    }
    /*</remove>*/
}
